connection_test.go

```diff
@@ -1079,3 +1084,104 @@ func TestContextCanceledOnTCPClose(t *testing.T) {
 		clientCh.Close()
 	})
 }
+
+// getConnection returns the introspection result for the unique inbound or
+// outbound connection. An assert will be raised if there is more than one
+// connection of the given type.
+func getConnection(t *testing.T, ch *Channel, direction int) *ConnectionRuntimeState {
+	state := ch.IntrospectState(&IntrospectionOptions{
+		IncludeEmptyPeers: false,
+	})
+
+	for _, peer := range state.RootPeers {
+		var connections []ConnectionRuntimeState
+		if direction == inbound {
+			connections = peer.InboundConnections
+		} else {
+			connections = peer.OutboundConnections
+		}
+
+		assert.True(t, len(connections) <= 1, "Too many connections found")
```

Comment:
use `assert.Len` instead of `assert.True` so you get more useful information on a failure

Reply:
Not the right choice here, because len(connections) == 0 is also valid. The code here also handles the case where a relay has 2 peers, and we're only interested in the inbound direction.

Reply:
i see, in that case can you add the list of connections found in the error message.
```
assert.True(t, len(connections) <= 1, "Too many connections found: %+v", connections)
```

Reply:
Done